packages/cadl-ranch-specs/http/versioning/removed/main.tsp

```diff
@@ -123,3 +144,38 @@ interface InterfaceV1 {
   @route("/v1")
   v1InInterface(@body body: ModelV1): ModelV1;
 }
+
+/** This operation should be generated with the signatures of both the latest and the original versions, rather than beta version */
+@scenario
+@scenarioDoc("""
+  path: "/versioning/removed/api-version:v1/v3"
+  Expected request body:
+  ```json
+  { "id": "123", "enumProp": "enumMemberV1" }
+  ```
+  """)
+@post
+@route("/v3inversion1")
+op modelV3InVersion1(@body body: ModelV3): ModelV3;
+
+@scenario
+@scenarioDoc("""
+  Expected request body:
+  ```json
+  { "id": "123", "enumProp": "enumMemberBeta"}
+  ```
+  """)
+@post
+@route("/v3inversionbeta")
+op modelV3InVersionBeta(@body body: ModelV3): ModelV3;
+
+@scenario
+@scenarioDoc("""
+  Expected request body:
+  ```json
+  { "id": "123", "enumProp": "enumMemberV1" }
+  ```
+  """)
+@post
+@route("/v3inversion2")
+op modelV3InVersion2(@body body: ModelV3): ModelV3;
```

Comment:
we should not add three op to test three version, instead, we should add three tests in the mock api.

Reply:
In actual operations, the removed mockapi seems to recognize these tests more based on the expected body rather than the path. Therefore, for the BetaVersion mockapi, it is difficult to merge it with the other two V1 and V2 versions of the mockapi. The generated code is based on the V2 version, so in actual generation, if we remove a property in the Beta version, it will also be absent in the V2 version. Therefore, we can only handle it again using @added.

Reply:
@tadelesh Is the current version of mockapi a suitable solution?